Add Order class and its associates

rntsoma opened the pull request:
Add Order class, Order repository and Order resources
The method to return the Order is a GET
From the project requisites, the Order contains:
    1. id
    2. A client id
    3. A list of products that the client ordered
Other attributes can be added later on

Currently, since there are no orders available at the database,
the query will return an empty list

rntsoma committed:
Add Order class and its associates

Add Order class, Order repository and Order resources
The method to return the Order is a GET
From the project requisites, the Order contains:
    1. id
    2. A client id
    3. A list of products that the client ordered
Other attributes can be added later on

Currently, since there are no orders available at the database,
the query will return an empty list

hdalmora commented:
Na var de instância productsList do model Order vc setou uma Lista de String, mas não seria melhor receber uma lista de produtos, no caso List(Product)? Quando o cliente seleciona um item na lista de um estabelecimento específico vamos adicionando cada produto nessa lista, e depois ficaria mais fácil de calcular o preço final do pedido, etc. 

Veja se vcs concordam nesse caso.

hdalmora commented:
Pensei também no caso do carrinho. Devemos ter uma classe para gerenciar o carrinho do cliente, pois só iremos gerar uma Order quando o cliente finalizar o carrinho e o pagamento (ou o pedido para pagar no ato da entrega). Então a classe Order poderia receber como um atributo o proprio carrinho, que já ira conter os produtos, o valor total pago (ou a ser pago - forma de pagamento)

rntsoma commented:
> Na var de instância productsList do model Order vc setou uma Lista de String, mas não seria melhor receber uma lista de produtos, no caso List(Product)? Quando o cliente seleciona um item na lista de um estabelecimento específico vamos adicionando cada produto nessa lista, e depois ficaria mais fácil de calcular o preço final do pedido, etc.
> 
> Veja se vcs concordam nesse caso.

No caso, eu tinha pensado em criar a lista baseado apenas nos ID's dos produtos, por isso uma lista de strings. Daria para mudar para Produto e concordo que facilitaria algumas lógicas associadas, vou mudar para seguir sua sugestão.

hdalmora commented:
> > Na var de instância productsList do model Order vc setou uma Lista de String, mas não seria melhor receber uma lista de produtos, no caso List(Product)? Quando o cliente seleciona um item na lista de um estabelecimento específico vamos adicionando cada produto nessa lista, e depois ficaria mais fácil de calcular o preço final do pedido, etc.
> > Veja se vcs concordam nesse caso.
> 
> No caso, eu tinha pensado em criar a lista baseado apenas nos ID's dos produtos, por isso uma lista de strings. Daria para mudar para Produto e concordo que facilitaria algumas lógicas associadas, vou mudar para seguir sua sugestão.

Legal. Criei um card no Trello para a questão do carrinho de compras. Se achar viável a idéia podemos discutir lá, se vc quiser puxar o card pra vc. Valeu Renato.

rntsoma committed:
Change list from String to Product

As discussed, it would be better to have a list of Products instead
of a list of ID's.
